CMakeLists.txt

```
@@ -81,6 +81,7 @@ function(packageProject)
       endif()
 
       string(TOUPPER ${PROJECT_NAME} UPPERCASE_PROJECT_NAME)
+      string(REGEX REPLACE [^a-zA-Z0-9] _ UPPERCASE_PROJECT_NAME ${UPPERCASE_PROJECT_NAME})
```

I think it would be a good idea to add a comment here, explaining why we remove non alphanumeric characters.

```suggestion
      # ensure that the generated macro does not include invalid characters
      string(REGEX REPLACE [^a-zA-Z0-9] _ UPPERCASE_PROJECT_NAME ${UPPERCASE_PROJECT_NAME})
```